### DISCUSSION TITLE
Getting the last commit message into the run command

### DISCUSSION THREAD

**Discussion by KoalaTeaSoftware** [community/community#27193] [Actions]
<p>In my Java project, there is  a properties file that is used by a plugin to populate a box on an output HTML file (making the result of my Selenium test look nice).</p>
<p>I just wanted to take the text of the last commit to the subject repo and use sed to hack the relevant property in the file. I have been able to make a similar hack using a static string, but using dynamically generated data (e.g. `git log -1 --pretty=format::"%s"’) leaves me cold.</p>
<p>I nearly got there with set-env, but that is deprecated.</p>
<p>Can anyone please tell me how this can be done?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="KoalaTeaSoftware" data-post="1" data-topic="148913">
<div class="title">
<div class="quote-controls"></div>
 KoalaTeaSoftware:</div>
<blockquote>
<p>I nearly got there with set-env, but that is deprecated.</p>
</blockquote>
</aside>
<p>It has been replaced by the <code>GITHUB_ENV</code> <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files">environment file</a>. The other thing you need to know is that in Bash you can use <code>$(command)</code> to substitute the output of <code>command</code>. So to set a <code>COMMIT_MSG</code> environment variable for use in later steps:</p>
<pre><code class="lang-yaml">- name: get commit message
  run: |
    echo "COMMIT_MSG=$(git log -1 --pretty=format:%s)" &gt;&gt;$GITHUB_ENV
</code></pre>
<p>Or if you need the output only for the <code>sed</code> command you could skip the variable and just do something like:</p>
<pre><code class="lang-yaml">- name: substitute log message
  run: |
   sed -e "s/@COMMIT_MSG@/$(git log -1 --pretty=format:%s)/" project.properties.in &gt;project.properties
</code></pre>

**Comment by KoalaTeaSoftware** [comment]
<p>Excellent! Thanks for this, I now can achieve what I was trying to do.</p>
<p>I have now realised that what was doing was, actually, not going to achieve what I actually wanted doing, but that is The Human Condition.</p>
